14-tidyr.Rmd

```diff
@@ -0,0 +1,154 @@
+---
+layout: page
+title: R for reproducible scientific analysis
+subtitle: Dataframe manipulation with tidyr
+minutes: 60
+---
+
+```{r, include=FALSE}
+source("tools/chunk-options.R")
+opts_chunk$set(fig.path = "fig/12-plyr-")
+# Silently load in the data so the rest of the lesson works
+gapminder <- read.csv("data/gapminder-FiveYearData.csv", header=TRUE)
+```
+
+> ## Learning objectives {.objectives}
+>
+> * To be able to convert dataframes between 'long' and 'wide' format with `tidyr`
+>
+
+Researchers often want to manipulate their data from the 'wide' to the 'long' format, or vice-versa. The 'long' format is where:
+
+ - each column is a variable
+ - each row is an observation
+
+In the 'long' format, you usually have 1 column for the observed variable and the other columns are ID variables. 
+
+For the 'wide' format each row is often a site/subject/patient and you have multiple observation variables. These can be either repeated observations over time, or observation of multiple variables (or a mix of both). You may find data input may be simpler or some other applications may prefer the 'wide' format. However, many of `R`'s functions have been designed assuming you have 'long' format data. This tutorial will help you efficiently transform your data regardless of original format.
+ 
+> draw this on the board? I'd like to include a diagram here too
+ 
+###Getting started
+First install the packages if you haven't already done so (you probably installed dplyr in the previous lesson):
+```{r,eval=FALSE}
+install.packages("tidyr")
+install.packages("dplyr")
+```
+
+Load the packages
+```{r,message=FALSE}
+library("tidyr")
+library("dplyr")
+```
+
+First, lets look at the structure of our gapminder dataframe:
+```{r}
+str(gapminder)
+```
+
+It is somewhere in between the purely 'long' and 'wide' format. We have 3 "ID variables" (`continent`, `country`, `year`) and 3 "Observation variables" (`pop`,`lifeExp`,`gdpPercap`). I usually prefer my data in this format in most cases despite not having ALL observations in 1 column. Since all 3 observation variables have different units, there are not too many operations that would need us to stretch out this dataframe any longer (i.e. 4 ID variables and 1 Observation variable). Many or R's vector based operations like oav(), lm(), or glm() will be easiest in this intermediate format (i.e. long format for each dependent variable, but not the longest format).
```

Comment:
Not sure many novice learners will have encountered these modeling functions. Maybe either worth spending a little more time explaining the type of function that prefer this intermediate formats or else remove that to keep focused?